[untested] Add monochrome menubar icon for mac

felixbuenemann opened the pull request:
This should align the icon with Apple design guidelines and work with both dark and light menu bars.

It contains a 2 Bit version of the icon in 44x44 px (Retina) created from the SVG version of the logo and sets this as a mask image, which requires at least QT 5.6.

If this works as intended it should negate the icon for dark mode menu bars.

Fixes #1240

felixbuenemann commented:
I haven't been able to test this due to lack of build instructions for macOS in the README.

Saviq commented:
Cool, thanks @felixbuenemann!

Only yesterday I filed canonical-web-and-design/multipass-design#47, this may make this quite a bit easier :)

bors try

felixbuenemann commented on src/client/gui/gui_cmd.cpp:
```diff
@@ -264,7 +264,12 @@ void cmd::GuiCmd::create_menu()
+    tray_icon.setIcon(QIcon{":images/multipass-menubar-icon-mac.png"});
+#else
     tray_icon.setIcon(QIcon{":images/multipass-icon.png"});
+#endif
```
Thanks, I've incorporated the changes.

felixbuenemann commented:
Please do not merge yet, I found a simple way to test the icon using PySide:

```python
from PySide2 import QtCore, QtWidgets, QtGui

app = QtWidgets.QApplication(["tray_icon_test"])
app.setAttribute(QtCore.Qt.AA_UseHighDpiPixmaps)
app.setQuitOnLastWindowClosed(False)

icon = QtGui.QIcon("src/client/gui/resources/images/multipass-menubar-icon-mac.png")
icon.setIsMask(True)
tray_icon = QtWidgets.QSystemTrayIcon(icon)

tray_icon.show()

app.exec_()
```

It works fine and looks good on a retina display, but on nin-retina the linex of the "M" become to thin, so I'll need to tweak the icon a bit to make them wider.

felixbuenemann commented:
Btw. it looks like the multipass-icon-png is overcropped by a pixel or to at the top and left, so it isn't completely circular. I can try to fix that while I'm at it.

felixbuenemann commented:
I've increased the size of the mac menubar icon by about 2px which improves legibility on non-retina displays. I didn't touch the width of the strokes, since it looks OK now.

I fixed the cropped edges on the color multipass-icon.png in another commit, see the rich diff of commit 644f9e1cf87f2602bb00ea4b53fd802e2074e3ea to see the difference.

Saviq commented:
Thanks Felix!

bors try

Saviq commented:
bors try

Saviq commented:
Hi @felixbuenemann, I tested the result and it's definitely looking better (see https://github.com/canonical-web-and-design/multipass-design/issues/47#issuecomment-609630437), but there's some love that the icon could receive, especially in low-DPI dark mode.

I'll keep this PR around for now and see if our design team can come up with something :)

felixbuenemann commented:
@Saviq The multipass-design repo is private, so I can't see it.

It should pe possible to increase the stroke of the lines of the M to increase legibility, but since I worked off-of the `data/multipass.gui.svg` where the dots and the lines are already one shape, I couldn't figure out a good way to do it consistently.

It might be easy for someone who works with Illustrator on a daily basis, but I'm primarily a programmer with some design knowledge.

felixbuenemann commented:
Some hints on the workflow might be helpful for your designers:

I first made the icon black and used the "M" shape as a cutout, so that the background was translucent. Then I changed the artboard origin to 0,0 and tweaked the arboard size to a multiple of 2 and until there was no cut-off at the edges when doing a "Save For Web (legacy)" to an 88x88 Px PNG.

The multiple of two on the artboard is important, or the logo will never be perfectly centered.

I opened the resulting 32-Bit PNG in Photoshop, converted it to Grayscale mode and saved it to a new file (using Save As, not Web export), which results in a 2-Bit PNG with alpha channel.

As a last step I used pngcrush with the `-rem allb -noreduce -blacken -brute` to optimize the file size and get rid of unneeded metadata. A similar result could be achieved with `optipng -o7 -zm1-9 -nx -strip all`, but make sure to disable all reductions, or the resulting file will be converted to 8-Bit indexed with alpha channel, making it RGBA instead of grayscale.

envygeeks commented:
Is there anything I can do to help push this along? I'm happy to jump into Illustrator or Sketch and fix whatever needs fixing if it pushes it closer to release since the icon really causes my eye to twitch.

willshen commented:
anything that I can do to help here? Would love to see this through!

townsend2010 commented:
Hey all,

We are in the process of getting some monochrome Multipass assets from our Design team. Unfortunately, I don't have an ETA on when we will get them.

willshen commented:
Thanks for the update @townsend2010 - will love to see this long running PR finally merged!

townsend2010 commented:
@willshen,

Yes, me too :wink:

andrei-toterman commented:
This was resolved in #3282 as part of moving to the new Flutter GUI.
